server/api/impl/aws/provision_data.go

```diff
@@ -0,0 +1,175 @@
+
+			data, err := ioutil.ReadAll(manifestFile)
+			if err != nil {
+				return nil, errors.Wrap(err, "can't read manifest.json file")
+			}
+			if err := json.Unmarshal(data, &mani); err != nil {
+				return nil, errors.Wrapf(err, "can't unmarshal manifest.json file %s", string(data))
+			}
+		case strings.HasSuffix(file.Name, ".zip"):
+			lambdaFunctionFile, err := file.Open()
+			if err != nil {
+				return nil, errors.Wrapf(err, "can't open file %s", file.Name)
+			}
+			defer lambdaFunctionFile.Close()
+
+			bundleFunctions = append(bundleFunctions, FunctionData{
+				Name:   strings.TrimSuffix(file.Name, ".zip"),
+				Bundle: lambdaFunctionFile,
+			})
+		case strings.HasPrefix(file.Name, bundleStaticAssetsFolder):
+			assetName := strings.TrimPrefix(file.Name, bundleStaticAssetsFolder)
+			if assetName == "" {
+				continue
+			}
+			assetFile, err := file.Open()
+			if err != nil {
+				return nil, errors.Wrapf(err, "can't open file %s", file.Name)
+			}
+			defer assetFile.Close()
+
+			assets = append(assets, AssetData{
+				Key:  assetName,
+				File: assetFile,
+			})
+			c.logger.Debug("Found function bundle", "file", file.Name)
+		default:
+			c.logger.Info("Unknown file found in app bundle", "file", file.Name)
+		}
+	}
+
+	if mani == nil {
+		return nil, errors.New("no manifest found")
+	}
+
+	resFunctions := []FunctionData{}
+
+	// Matching bundle functions to the functions listed in manifest
+	// O(n^2) code for simplicity
+	for _, bundleFunction := range bundleFunctions {
+		for _, manifestFunction := range mani.Functions {
+			if strings.HasSuffix(bundleFunction.Name, manifestFunction.Name) {
+				resFunctions = append(resFunctions, FunctionData{
+					Bundle:  bundleFunction.Bundle,
+					Name:    manifestFunction.Name,
+					Handler: manifestFunction.Handler,
+					Runtime: manifestFunction.Runtime,
+				})
+				continue
+			}
+		}
+	}
+
+	assetMap := generateAssetNames(mani, assets)
+	functionMap, err := generateFunctionNames(mani, resFunctions)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ProvisionData{
+		S3BucketName:    c.AppsS3Bucket,
+		StaticFiles:     assetMap,
+		LambdaFunctions: functionMap,
+		Manifest:        mani,
+	}, nil
+}
+
+func generateAssetNames(manifest *apps.Manifest, assets []AssetData) map[string]AssetData {
+	assetMap := make(map[string]AssetData, len(assets))
+	for _, asset := range assets {
+		assetMap[asset.Key] = AssetData{
+			Key:  GetAssetFileKey(manifest.AppID, manifest.Version, asset.Key),
+			File: asset.File,
+		}
+	}
+	return assetMap
+}
+
+func generateFunctionNames(manifest *apps.Manifest, functions []FunctionData) (map[string]FunctionData, error) {
+	functionMap := make(map[string]FunctionData, len(functions))
+	for _, function := range functions {
+		name, err := getFunctionName(manifest.AppID, manifest.Version, function.Name)
+		if err != nil {
+			return nil, errors.Wrap(err, "can't get function name")
+		}
+		functionMap[function.Name] = FunctionData{
```

Comment:
Is this the (TerraForm) standard way of packaging these AWS assets/artifacts for pipeline consumption - or is it an internal contract between us and the cloud team? If it's the first option, can we reference some TF documentation for future reference?